### PR TITLE
[8.13][cloud] CNVM Findings page enhancements 

### LINKED ISSUE #4746: [Request] Updates Vulnerabilities Findings Page
### Description

This is an extension of grouping and table customisation options introduced in 8.12 for CSPM/KSPM Findings. This is to introduce similar features in CNVM Findings Page.

### Background & resources

* PRs: 
* Issues/metas: 
- https://github.com/elastic/kibana/issues/165788
- https://github.com/elastic/kibana/issues/169050
- https://github.com/elastic/kibana/issues/172197
- https://github.com/elastic/kibana/issues/169052
- https://github.com/elastic/kibana/issues/169048
- https://github.com/elastic/kibana/issues/173405
- https://github.com/elastic/kibana/issues/175574

* Point of contact: @opauloh @Omolola-Akinleye 
* Test environments: 

### Which documentation set does this change impact?

ESS and serverless

### ESS release

8.13

### Serverless release

April 1, 2024

### Feature differences

Identical

### API docs impact

No

### Prerequisites, privileges, feature flags

- License subscription access to Cloud Security
- User with a Role with read privilege to the following indices: `logs-cloud_security_posture.scores-default`, `logs-cloud_security_posture.findings_latest-default`, `logs-cloud_security_posture.vulnerabilities_latest-default`

### PR DESCRIPTION
**Pull request by benironside**
Fixes #4746 by updating the CNVM Findings page documentation to reflect the latest improvements, which include improved grouping capabilities, a full implementation of EUI grid (I _think_ that's what the component is called - anyway the table now has more features), and other UI changes.

Previews:
- [CNVM Findings page](https://security-docs_bk_4937.docs-preview.app.elstc.co/guide/en/security/master/vuln-management-findings.html) (most of the changes are here)
- Findings pages for the [CSPM](https://security-docs_bk_4937.docs-preview.app.elstc.co/guide/en/security/master/cspm-findings-page.html) and [KSPM](https://security-docs_bk_4937.docs-preview.app.elstc.co/guide/en/security/master/findings-page.html) integrations (minor changes to maintain consistency across these related pages)

### CODE REVIEW

File: docs/cloud-native-security/cspm-findings.asciidoc [hunk 1]
```diff
@@ -37,7 +39,7 @@ You can filter findings data in two ways:
 [discrete]
 [[cspm-customize-the-findings-table]]
 === Customize the Findings table
-Use the toolbar buttons in the upper-left of the Findings table to customize the columns you want displayed:
+When grouping is disabled, you can use the toolbar buttons in the upper-left of the Findings table to select which columns appear:
```
Comment (opauloh) [inline docs/cloud-native-security/cspm-findings.asciidoc:42]: Is not necessarily when grouping is disabled, users can still customize the table in the grouped view as long the table is visible

File: docs/cloud-native-security/kspm-findings.asciidoc [hunk 2]
```diff
@@ -21,10 +20,12 @@ By default, the Findings page lists all findings, without grouping or filtering.
 [discrete]
 === Group findings
 
-. Click **Group findings by:** to open a list of fields by which you can group findings. Select one of the suggested group-by fields, or click **Custom field** to choose your own group-by field. 
-. When grouping is turned on, click a group to expand it and examine all findings within that group.
+. Click **Group vulnerabilities by:** to open a list of fields by which you can group your data. Select one of the suggested fields, or click **Custom field** to choose your own. You can select up to three group-by fields at once.
```
Comment (opauloh) [inline docs/cloud-native-security/kspm-findings.asciidoc:23]: Group vulnerabilities by -> Group findings by

File: docs/cloud-native-security/kspm-findings.asciidoc [hunk 3]
```diff
@@ -21,10 +20,12 @@ By default, the Findings page lists all findings, without grouping or filtering.
 [discrete]
 === Group findings
 
-. Click **Group findings by:** to open a list of fields by which you can group findings. Select one of the suggested group-by fields, or click **Custom field** to choose your own group-by field. 
-. When grouping is turned on, click a group to expand it and examine all findings within that group.
+. Click **Group vulnerabilities by:** to open a list of fields by which you can group your data. Select one of the suggested fields, or click **Custom field** to choose your own. You can select up to three group-by fields at once. 
+. When grouping is turned on, click a group to expand it and examine all sub-groups or findings within that group.
 . To turn off grouping, click **Group findings by:** and select **None**.
 
+NOTE: Multiple groupings apply to your data in the order you selected them; for example if you first select **Cloud account**, and then select **Resource**, the top-level grouping will be based on **Cloud account**, and its subordinate grouping will be based on **Resource**.
```
Comment (opauloh) [inline docs/cloud-native-security/kspm-findings.asciidoc:27]: for a better example closer to the real scenario I would suggest you replace **Cloud account** with **Kubernetes cluster** (for the KSPM docs only):


NOTE: Multiple groupings apply to your data in the order you selected them; for example if you first select **Kubernetes cluster**, and then select **Resource**, the top-level grouping will be based on **Kubernetes cluster**, and its subordinate grouping will be based on **Resource**.

File: docs/cloud-native-security/kspm-findings.asciidoc [hunk 4]
```diff
@@ -36,7 +37,7 @@ You can filter findings data in two ways:
 [discrete]
 [[kspm-customize-the-findings-table]]
 === Customize the Findings table
-Use the toolbar buttons in the upper-left of the Findings table to customize the columns you want displayed:
+When grouping is disabled, you can use the toolbar buttons in the upper-left of the Findings table to select which columns appear:
```
Comment (opauloh) [inline docs/cloud-native-security/kspm-findings.asciidoc:40]: same consideration as above

File: docs/cloud-native-security/vuln-management-findings.asciidoc [hunk 5]
```diff
@@ -1,32 +1,54 @@
...
 
-When *Group by* is set to *None*, you can sort the Findings table by clicking the column headings or the *Sort fields* button to the upper left of the table. When sorting is active, the *Sort fields* button changes to *X fields sorted* (where _X_ is the number of fields sorting your data), and can be used to modify or clear sorting.
+* *The KQL search bar*: For example, search for `vulnerability.severity : "HIGH"` to view high severity vulnerabilities.
+* *In-table value filters*: Hover over a finding to display available inline actions. Use the Filter In (plus) and Filter Out (minus) buttons.
 
-Independent of grouping, you can filter data in two ways:
+[discrete]
+[[cnvm-customize-the-findings-table]]
+=== Customize the Findings table
+When grouping is disabled, you can use the toolbar buttons in the upper-left of the Findings table to select which columns appear:
```
Comment (opauloh) [inline docs/cloud-native-security/vuln-management-findings.asciidoc:45]: same consideration of grouping not need to be disabled

**Review by opauloh — DISMISSED** [review]
just a couple of minor issues and an improvement suggestion

File: docs/cloud-native-security/cspm-findings.asciidoc [hunk 6]
```diff
@@ -1,7 +1,7 @@
 [[cspm-findings-page]]
 = Findings page
 
-The Findings page displays the configuration risks identified by the <<cspm,CSPM>> and <<kspm,KSPM>> integrations.
+The Misconfigurations tab of the Findings page displays the configuration risks identified by the <<cspm,CSPM>> and <<kspm,KSPM>> integrations.
```
Comment (nastasha-solomon) [inline docs/cloud-native-security/cspm-findings.asciidoc:4]:
```suggestion
The **Misconfigurations** tab of the Findings page displays the configuration risks identified by the <<cspm,CSPM>> and <<kspm,KSPM>> integrations.
```

File: docs/cloud-native-security/cspm-findings.asciidoc [hunk 7]
```diff
@@ -21,10 +21,12 @@ By default, the Findings page lists all findings, without grouping or filtering.
 [discrete]
 === Group findings 
 
-. Click **Group findings by:** to open a list of fields by which you can group findings. Select one of the suggested group-by fields, or click **Custom field** to choose your own group-by field. 
-. When grouping is turned on, click a group to expand it and examine all findings within that group.
+. Click **Group findings by:** to open a list of fields by which you can group your data. Select one of the suggested fields, or click **Custom field** to choose your own. You can select up to three group-by fields at once. 
+. When grouping is turned on, click a group to expand it and examine all sub-groups or findings within that group.
 . To turn off grouping, click **Group findings by:** and select **None**.
 
+NOTE: Multiple groupings apply to your data in the order you selected them; for example if you first select **Cloud account**, and then select **Resource**, the top-level grouping will be based on **Cloud account**, and its subordinate grouping will be based on **Resource**.
```
Comment (nastasha-solomon) [inline docs/cloud-native-security/cspm-findings.asciidoc:28]:
```suggestion
NOTE: Multiple groupings apply to your data in the order you selected them. For example, if you first select **Cloud account**, and then select **Resource**, the top-level grouping will be based on **Cloud account**, and its subordinate grouping will be based on **Resource**. 
```

File: docs/cloud-native-security/kspm-findings.asciidoc [hunk 8]
```diff
@@ -1,7 +1,7 @@
 [[findings-page]]
 = Findings page
 
-The Findings page displays the configuration risks identified by the <<cspm,CSPM>> and <<kspm,KSPM>> integrations.
+The Misconfigurations tab of the Findings page displays the configuration risks identified by the <<cspm,CSPM>> and <<kspm,KSPM>> integrations.
```
Comment (nastasha-solomon) [inline docs/cloud-native-security/kspm-findings.asciidoc:4]:
```suggestion
The **Misconfigurations** tab of the Findings page displays the configuration risks identified by the <<cspm,CSPM>> and <<kspm,KSPM>> integrations.
```

File: docs/cloud-native-security/kspm-findings.asciidoc [hunk 9]
```diff
@@ -21,10 +20,12 @@ By default, the Findings page lists all findings, without grouping or filtering.
 [discrete]
 === Group findings
 
-. Click **Group findings by:** to open a list of fields by which you can group findings. Select one of the suggested group-by fields, or click **Custom field** to choose your own group-by field. 
-. When grouping is turned on, click a group to expand it and examine all findings within that group.
+. Click **Group findings by:** to open a list of fields by which you can group your data. Select one of the suggested fields, or click **Custom field** to choose your own. You can select up to three group-by fields at once. 
+. When grouping is turned on, click a group to expand it and examine all sub-groups or findings within that group.
 . To turn off grouping, click **Group findings by:** and select **None**.
 
+NOTE: Multiple groupings apply to your data in the order you selected them; for example if you first select **Kubernetes cluster**, and then select **Resource**, the top-level grouping will be based on **Kubernetes cluster**, and its subordinate grouping will be based on **Resource**.
```
Comment (nastasha-solomon) [inline docs/cloud-native-security/kspm-findings.asciidoc:27]:
```suggestion
NOTE: Multiple groupings apply to your data in the order you selected them. For example, if you first select **Kubernetes cluster**, and then select **Resource**, the top-level grouping will be based on **Kubernetes cluster**, and its subordinate grouping will be based on **Resource**. 
```

File: docs/cloud-native-security/vuln-management-findings.asciidoc [hunk 10]
```diff
@@ -1,32 +1,54 @@
 [[vuln-management-findings]]
-= Findings
+= Findings page
 
-The vulnerabilities findings page displays the vulnerabilities detected by the <<vuln-management-overview, CNVM integration>>. CNVM findings include metadata such as the CVE identifier, CVSS score, severity, affected package, and fix version if available, as well as information about impacted systems.
+The Vulnerabilities tab of the Findings page displays the vulnerabilities detected by the <<vuln-management-overview, CNVM integration>>.
```
Comment (nastasha-solomon) [inline docs/cloud-native-security/vuln-management-findings.asciidoc:4]:
```suggestion
The **Vulnerabilities** tab of the Findings page displays the vulnerabilities detected by the <<vuln-management-overview, CNVM integration>>. 
```

File: docs/cloud-native-security/vuln-management-findings.asciidoc [hunk 11]
```diff
@@ -1,32 +1,54 @@
...
+
+[discrete]
+=== Group findings
+
+. Click **Group vulnerabilities by:** to open a list of fields by which you can group your data. Select one of the suggested fields, or click **Custom field** to choose your own. You can select up to three group-by fields at once. 
+. When grouping is turned on, click a group to expand it and examine all sub-groups or findings within that group.
+. To turn off grouping, click **Group findings by:** and select **None**.
 
-You can group your data by resource by selecting *Resource* from the *Group by* menu. When data is grouped by resource, you can click on the name of a virtual machine to view all vulnerabilities that were found on it.
+NOTE: Multiple groupings apply to your data in the order you selected them; for example if you first select **Cloud account**, and then select **Resource**, the top-level grouping will be based on **Cloud account**, and its subordinate grouping will be based on **Resource**, as in the following screenshot:
```
Comment (nastasha-solomon) [inline docs/cloud-native-security/vuln-management-findings.asciidoc:30]: Minor tweak to the last sentence for clarity.

```suggestion
NOTE: Multiple groupings apply to your data in the order you selected them. For example, if you first select **Cloud account**, and then select **Resource**, the top-level grouping will be based on **Cloud account**, and its subordinate grouping will be based on **Resource**, as demonstrated in the following screenshot:
```

File: docs/cloud-native-security/vuln-management-findings.asciidoc [hunk 12]
```diff
@@ -1,32 +1,54 @@
...
 image::images/cnvm-findings-grouped.png[The Vulnerabilities tab of the Findings page]
 
+[discrete]
+[[cnvm-filter-findings]]
+=== Filter findings
+You can filter the data in two ways:
 
-When *Group by* is set to *None*, you can sort the Findings table by clicking the column headings or the *Sort fields* button to the upper left of the table. When sorting is active, the *Sort fields* button changes to *X fields sorted* (where _X_ is the number of fields sorting your data), and can be used to modify or clear sorting.
+* *The KQL search bar*: For example, search for `vulnerability.severity : "HIGH"` to view high severity vulnerabilities.
+* *In-table value filters*: Hover over a finding to display available inline actions. Use the Filter In (plus) and Filter Out (minus) buttons.
```
Comment (nastasha-solomon) [inline docs/cloud-native-security/vuln-management-findings.asciidoc:40]:
```suggestion
* *In-table value filters*: Hover over a finding to display available inline actions. Use the **Filter In** (plus) and **Filter Out** (minus) buttons.
```

File: docs/cloud-native-security/vuln-management-findings.asciidoc (hunk 5 above)
Comment (nastasha-solomon) [inline docs/cloud-native-security/vuln-management-findings.asciidoc:45]:
```suggestion
When grouping is turned off, you can use the toolbar buttons in the upper-left of the Findings table to select which columns appear:
```

**Review by nastasha-solomon — DISMISSED** [review]
Left some minor suggestions for your consideration. Looks great overall!

**Review by opauloh — APPROVED** [review]
it looks good, thanks for the changes

File: docs/cloud-native-security/cspm-findings.asciidoc [hunk 13]
```diff
@@ -21,10 +21,12 @@ By default, the Findings page lists all findings, without grouping or filtering.
 [discrete]
 === Group findings 
 
-. Click **Group findings by:** to open a list of fields by which you can group findings. Select one of the suggested group-by fields, or click **Custom field** to choose your own group-by field. 
-. When grouping is turned on, click a group to expand it and examine all findings within that group.
+. Click **Group findings by:** to open a list of fields by which you can group your data. Select one of the suggested fields, or click **Custom field** to choose your own. You can select up to three group-by fields at once.
```
Comment (joepeeples) [inline docs/cloud-native-security/cspm-findings.asciidoc:24]: This step appears a few other times in the PR, so the same suggested edits apply there too.
```suggestion
. Click **Group findings by** to group your data by a field. Select one of the suggested fields or **Custom field** to choose your own. You can select up to three group-by fields at once. 
```

File: docs/cloud-native-security/cspm-findings.asciidoc [hunk 14]
```diff
@@ -21,10 +21,12 @@ By default, the Findings page lists all findings, without grouping or filtering.
 [discrete]
 === Group findings 
 
-. Click **Group findings by:** to open a list of fields by which you can group findings. Select one of the suggested group-by fields, or click **Custom field** to choose your own group-by field. 
-. When grouping is turned on, click a group to expand it and examine all findings within that group.
+. Click **Group findings by:** to open a list of fields by which you can group your data. Select one of the suggested fields, or click **Custom field** to choose your own. You can select up to three group-by fields at once. 
+. When grouping is turned on, click a group to expand it and examine all sub-groups or findings within that group.
 . To turn off grouping, click **Group findings by:** and select **None**.
```
Comment (joepeeples) [inline docs/cloud-native-security/cspm-findings.asciidoc:26]:
```suggestion
. To turn off grouping, click **Group findings by** and select **None**.
```

File: docs/cloud-native-security/vuln-management-findings.asciidoc [hunk 15]
```diff
@@ -1,32 +1,54 @@
...
 
 image::images/cnvm-findings-grouped.png[The Vulnerabilities tab of the Findings page]
 
+[discrete]
+[[cnvm-filter-findings]]
+=== Filter findings
+You can filter the data in two ways:
 
-When *Group by* is set to *None*, you can sort the Findings table by clicking the column headings or the *Sort fields* button to the upper left of the table. When sorting is active, the *Sort fields* button changes to *X fields sorted* (where _X_ is the number of fields sorting your data), and can be used to modify or clear sorting.
+* *The KQL search bar*: For example, search for `vulnerability.severity : "HIGH"` to view high severity vulnerabilities.
```
Comment (joepeeples) [inline docs/cloud-native-security/vuln-management-findings.asciidoc:39]:
```suggestion
* *KQL search bar*: For example, search for `vulnerability.severity : "HIGH"` to view high severity vulnerabilities.
```

**Review by joepeeples — COMMENTED** [review]
Couple of suggested edits, looks good! Ping again when you're ready for an approval.

File: docs/cloud-native-security/vuln-management-findings.asciidoc [hunk 16]
```diff
@@ -1,32 +1,54 @@
...
 
-Independent of grouping, you can filter data in two ways:
+[discrete]
+[[cnvm-customize-the-findings-table]]
+=== Customize the Findings table
+When grouping is turned off, you can use the toolbar buttons in the upper-left of the Findings table to select which columns appear:
 
-- *The KQL search bar*: Use this to filter your findings. For example, search for `vulnerability.id : CVE-2019-00001` to view all findings related to a particular vulnerability.
+* **Columns**: Select the left-to-right order in which columns appear.
+* **Sort fields** or **_n_ fields sorted**: Sort the table by one or more columns, or turn sorting off.
```
Comment (natasha-moore-elastic) [inline docs/cloud-native-security/vuln-management-findings.asciidoc:48]:
```suggestion
* **Sort fields**: Sort the table by one or more columns, or turn sorting off.
```
Looking at the updated screenshots, it looks like this copy has changed and it says **Sort fields _n_** even when sorting is on.

**Review by joepeeples — APPROVED** [review]
LGTM!

**Review by jmikell821 — COMMENTED** [review]
Just a few slight nits - looks good!

**Review by jmikell821 — APPROVED** [review]
LGTM!

**Pull request merged**